static/js/stream_sort.js

```diff
@@ -26,7 +26,11 @@ function compare_function(a, b) {
 
 export function sort_groups(streams, search_term) {
     const stream_id_to_name = (stream) => sub_store.get(stream).name;
-    streams = util.filter_by_word_prefix_match(streams, search_term, stream_id_to_name);
+    streams = util.filter_by_prefix_match_at_word_boundaries(
```

Comment:
I don't like this name that much. Do we need to write `word_boundary` in the function name? It feels too verbose.